src/GridFS/CollectionWrapper.php

```diff
@@ -253,6 +253,17 @@ public function insertFile($file): void
         $this->filesCollection->insertOne($file);
     }
 
+    /**
+     * Updates the filename field in the file document for all the files with a given filename.
+     */
+    public function updateFilenameForFilename(string $filename, string $newFilename): UpdateResult
+    {
+        return $this->filesCollection->updateMany(
```

Comment:
Ack - I think since renaming files by filename is not part of the spec and the spec only contains a suggested implementation **for users**, we should be fine adding this implementation. I'd suggest we amend the spec to add APIs to work on all revisions of a file (i.e. remove and rename by filename instead of by ID).

Reply:
> renaming files by filename is not part of the spec and the spec only contains a suggested implementation for users

@alcaeus: Good point. I overlooked that the instructions were telling _users_ to run the `rename()` operation for each ID, instead of discussing how a driver should implement filename-based renames.